src/main/kotlin/org/jitsi/jibri/service/impl/StreamingJibriService.kt

```diff
@@ -93,14 +94,19 @@ class StreamingJibriService(
             capturer.start(sink)
         }
 
-        jibriSelenium.addToPresence("session_id", streamingParams.sessionId)
-        jibriSelenium.addToPresence("mode", JibriIq.RecordingMode.STREAM.toString())
-        streamingParams.youTubeBroadcastId?.let {
-            if (!jibriSelenium.addToPresence("live-stream-view-url", "http://youtu.be/$it")) {
-                logger.error("Error adding live stream url to presence")
+        try {
+            jibriSelenium.addToPresence("session_id", streamingParams.sessionId)
+            jibriSelenium.addToPresence("mode", JibriIq.RecordingMode.STREAM.toString())
+            streamingParams.youTubeBroadcastId?.let {
+                if (!jibriSelenium.addToPresence("live-stream-view-url", "http://youtu.be/$it")) {
+                    logger.error("Error adding live stream url to presence")
+                }
             }
+            jibriSelenium.sendPresence()
+        } catch (t: Throwable) {
+            logger.error("Error while setting fields in presence", t)
+            publishStatus(ComponentState.Error(ErrorScope.SESSION, "Unable to set presence values"))
```

Comment:
Why did you prefer having start implement its own error handling versus having the caller of start implement it? Is it for the publishStatus behavior being the source of service status truth?

Reply:
yeah, pretty much.  starts are async now, so all status is pushed via publishStatus